src/minuteman_route_mgr.erl

```diff
@@ -162,71 +168,77 @@ code_change(_OldVsn, State, _Extra) ->
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
-
-get_routes(Pid) ->
+get_minuteman_iface(Pid) ->
+    {ok, Raw} = gen_netlink_client:rtnl_request(Pid, getlink, [match, root],
+                                                {inet, arphrd_none, 0, [], [], []}),
+    Ifaces = [link_msg_index(Msg) || #rtnetlink{msg = Msg} <- Raw,
+                                     link_msg_ifname(Msg) == ?MINUTEMAN_IFACE],
+    get_iface(Ifaces).
+
+get_iface([]) ->
+    lager:error("Missing minuteman interface, make sure 'ip link add ~p type dummy'" ++
+                " was run at service startup.", [?MINUTEMAN_IFACE]),
+    {error, "Missing minuteman interface."};
+get_iface([Iface]) ->  Iface;
+get_iface(Ifaces) ->
+    lager:error("Found duplicate ~p interfaces, this should never happen." ++
+                " Check your network interfaces for: ~p", [?MINUTEMAN_IFACE, [Ifaces]]),
+    {error, "Duplicate minuteman interface."}.
+
+%% see netlink.hrl for the element position
+link_msg_ifname(Msg) -> proplists:get_value(ifname, element(6, Msg)).
+link_msg_index(Msg) -> element(3, Msg).
+
+get_routes(Pid, Iface) ->
```

Comment:
You could fetch all the routes for an interface. Kernel will do the filtering for you